EasyLayout.Forms/EasyLayout.cs

```diff
@@ -567,112 +548,52 @@ private static int ConstantValue(Expression expr)
-            {
-                var mexpr = (MemberExpression)expr;
-                // todo: Does this even work, where the heck is MemberType?
-                if (mexpr.NodeType == ExpressionType.Constant)
-                {
+                case ConstantExpression _:
                     return true;
-                }
-                return false;
-            }
-
-            if (expr.NodeType == ExpressionType.Convert)
-            {
-                var cexpr = (UnaryExpression)expr;
-                return IsConstant(cexpr.Operand);
-            }
-
-            var methodCall = expr as MethodCallExpression;
-            if (methodCall != null)
-            {
-                return methodCall.Method.Name == nameof(ToConst);
+                case MemberExpression mexpr:
+                    // todo: Does this even work, where the heck is MemberType?
+                    return mexpr.NodeType == ExpressionType.Constant;
+                case UnaryExpression cexpr:
+                    return IsConstant(cexpr.Operand);
+                case MethodCallExpression methodCall:
+                    return methodCall.Method.Name == nameof(ToConst);
+                default:
+                    return false;
             }
-
-            return false;
         }
 
         private static LeftExpression ParseLeftExpression(Expression expr)
         {
-            Position? position = null;
-            MemberExpression viewExpr = null;
-
             var fExpr = expr as MethodCallExpression;
-            if (fExpr != null)
-            {
-                position = GetPosition(fExpr);
-                viewExpr = fExpr.Arguments.FirstOrDefault() as MemberExpression;
-            }
-
-            if (position == null)
-            {
-                var memExpr = expr as MemberExpression;
-                if (memExpr == null)
-                {
-                    throw new NotSupportedException("Left hand side of a relation must be a member expression, instead it is " + expr);
-                }
-
-                position = GetPosition(memExpr);
-
-                viewExpr = memExpr.Expression as MemberExpression;
-                viewExpr = viewExpr?.Expression as MemberExpression;
-            }
-
-            if (viewExpr == null)
-            {
-                throw new NotSupportedException("Constraints should use views's Top, Bottom, etc properties, or extension methods like GetCenter().  Did you forget to include '.Bounds' in your clause?");
-            }
-
-            var view = GetView(viewExpr);
-
+            var viewExpr = fExpr?.Arguments.FirstOrDefault() as MemberExpression;
             return new LeftExpression
             {
-                View = view,
-                Position = position.Value,
-                Name = viewExpr.Member.Name
+                View = GetView(viewExpr),
+                Position = GetPosition(fExpr),
+                Name = viewExpr?.Member.Name
```

Comment:
First of all, is it ok for viewExpr to be null?  I think not.  But if it is, isn't this now a null reference exception?